index.js

```diff
@@ -4259,6 +4319,189 @@ Agent.prototype.onMsg = function (buffer, rinfo) {
 	}
 };
 
+Agent.prototype.cast = function ( type, value ) {
+	switch (type) {
+		case ObjectType.Boolean:
+			return !! value;
+
+		case ObjectType.Integer:
+			if ( typeof value != "number" && typeof value != "string" ) {
+				throw new Error("Invalid Integer", value);
+			}
+			return typeof value == "number" ? value : parseInt(value, 10);
+
+		case ObjectType.OctetString:
+			if ( typeof value != "string" ) {
+				throw new Error("Invalid OctetString", value);
+			}
+			return value;
+
+		case ObjectType.OID:
+			if ( typeof value != "string" || ! value.match(/[0-9]+\([.][0-9]+\)+/) ) {
+				throw new Error("Invalid OID", value);
+			}
+			return value;
+
+		case ObjectType.Counter:
+		case ObjectType.Counter64:
+			// Counters should be initialized to 0 (RFC2578, end of section 7.9)
+			// We'll do so.
+			return 0;
+
+		case ObjectType.IpAddress:
+			// A 32-bit internet address represented as OCTET STRING of length 4
+			if ( typeof value != "string" || value.length != 4 ) {
+				throw new Error("Invalid IpAddress", value);
+			}
+			return value;
+
+		default :
+			// Assume the caller knows what he's doing
+			return value;
+	}
+};
+
+
+Agent.prototype.tryCreateInstance = function (varbind, requestType) {
+	var len;
+	var row;
+	var column;
+	var value;
+	var subOid;
+	var address;
+	var provider;
+	var providersByOid = {};
+	var providers = this.mib.providers;
+	var oid = varbind.oid;
+
+	// TODO: providersByOid should be generated only once. Either
+	// invalidate it upon each change to this.mib.providers, so that
+	// it's recreated as needed; or keep it in sync with
+	// this.mib.providers. It probably should be maintained as
+	// this.mib.providersByOid rather than local, here.
+	for (var name in providers) {
+		providersByOid[providers[name].oid] = providers[name];
+	}
+
+	// Look for the provider.
+	address = Mib.convertOidToAddress (oid);
+	len = address.length;
+	if ( len < 3 ) {
+		console.log("OID is fubar; don't try further");
+		return undefined;
+	}
+
+	// Extract the row and column number before we start searching
+	// upwards for the provider
+	row = address[len - 1];
+	column = address[len - 2];
```

Comment:
See comment below on this, but in short, the column calculation needs to be MibNode.getTableColumnFromInstanceNode()

and the row needs to be an integer array, returned by a new but similar call like:
MibNode.getRowIndexFromInstanceNode()